src/DynamoCoreWpf/Controls/StartPage.xaml.cs

```diff
@@ -417,6 +432,13 @@ private void OnStartPageLoaded(object sender, RoutedEventArgs e)
 
             var id = Wpf.Interfaces.ResourceNames.StartPage.Image;
             StartPageLogo.Source = dynamoViewModel.BrandingResourceProvider.GetImageSource(id);
+
+            /*if (dynamoViewModel.Model.PreferenceSettings.ShowGalleryAtStartUp)
+            {
+                dynamoViewModel.Model.PreferenceSettings.ShowGalleryAtStartUp = false;
+                dynamoViewModel.ShowGalleryCommand.Execute(null);
+            }*/
+
```

Comment:
Hi @Sancerio, as discussed this is still wrong, we shouldn't set `ShowGalleryAtStartUp = false` here, otherwise there is no way to implement *Always show gallery at start-up*. Instead, this is what I meant:

```csharp
private void DynamoView_Loaded(object sender, EventArgs e)
{
    // Before UsageReportingManager changes the value, back it up...
    var isFirstRun = dynamoViewModel.Model.PreferenceSettings.IsFirstRun;
    UsageReportingManager.Instance.CheckIsFirstRun(this, dynamoViewModel.BrandingResourceProvider);

    // Bla bla bla...


    // Pass it on to the start page initialization...
    InitializeStartPage(isFirstRun);
}

private void InitializeStartPage(bool isFirstRun)
{
    // Pass it on to the StartPageViewModel, so StartPageView can access it...
    startPage = new StartPageViewModel(dynamoViewModel, isFirstRun);
}
```

This way the value of `PreferenceSettings.IsFirstRun` can be changed by `UsageReportingManager` safely, `StartPageView` can still show the gallery on first run.